tests/storages_tests/test_journal.py

```diff
@@ -65,3 +66,32 @@ def test_concurrent_append_logs(log_storage_type: str) -> None:
 
         assert len(storage.read_logs(0)) == num_records
         assert all(record == r for r in storage.read_logs(0))
+
+
+def pop_waiting_trial(file_path: str, study_name: str) -> Optional[int]:
+    file_storage = optuna.storages.JournalFileStorage(file_path)
+    storage = optuna.storages.JournalStorage(file_storage)
+    study = optuna.load_study(storage=storage, study_name=study_name)
+    return study._pop_waiting_trial_id()
+
+
+def test_pop_waiting_trial_multiprocess_safe() -> None:
+    with tempfile.NamedTemporaryFile() as file:
```

Comment:
`STORAGE_MODES` variable, which is used with `pytest.mark.parametrize()`, contains `sqlite3`, `redis`, and `inmemory` except for `journal`.

* SQLite3 does not support a high-level of concurrency. It may raise "database is locked" error that makes tests fragile. We can increase the default timeout though.
* We use `fakeredis` for testing RedisStorage. It doesn't support multiple processes.

So here I put this test case instead of using `pytest.mark.parametrize("storage_mode", STORAGE_MODES)`.

Reply:
Oh I see. So sqlite3 cannot be used for multi-process optimization that includes enqueue trial.